python-sdk/src/astro/databases/snowflake.py

```diff
@@ -766,6 +762,49 @@ def _build_merge_sql(
         }
         return statement, params
 
+    def append_table(
+        self,
+        source_table: BaseTable,
+        target_table: BaseTable,
+        source_to_target_columns_map: dict[str, str],
+    ) -> None:
+        """
+        Append the source table rows into a destination table.
+        The argument `if_conflicts` allows the user to define how to handle conflicts.
+
+        :param source_table: Contains the rows to be appended to the target_table
+        :param target_table: Contains the destination table in which the rows will be appended
+        :param source_to_target_columns_map: Dict of source_table columns names to target_table columns names
+        """
+        target_table_sqla = self.get_sqla_table(target_table)
+        source_table_sqla = self.get_sqla_table(source_table)
+
+        target_columns: list[column]
+        source_columns: list[column]
+
+        if not source_to_target_columns_map:
+            target_columns = [Column(name=col, quote=True) for col in target_table_sqla.c.keys()]
+            source_columns = target_columns
+        else:
+            target_columns = [Column(name=col, quote=True) for col in source_to_target_columns_map.values()]
+            source_columns = [Column(name=col, quote=True) for col in source_to_target_columns_map.keys()]
+
+        sel = select(source_columns).select_from(source_table_sqla)
+        # TODO: We should fix the following Type Error
+        # incompatible type List[ColumnClause[<nothing>]]; expected List[Column[Any]]
+        sql = insert(target_table_sqla).from_select(target_columns, sel)  # type: ignore[arg-type]
+        self.run_sql(sql=sql)
+
+    @staticmethod
+    def get_merge_initialization_query(parameters: tuple) -> str:
+        """
+        Handles database-specific logic to handle constraints, keeping
+        it agnostic to database.
+        """
+        constraints = ",".join([f'"{p}"' for p in parameters])
+        sql = "ALTER TABLE {{table}} ADD CONSTRAINT airflow UNIQUE (%s)" % constraints
+        return sql
```

Comment:
Why are we introducing these methods as part of the capitalization PR? Is it because of the quoting? Doesn't the table identifier need quotes? It may be worth adding comments explaining why we have this and highlighting the underlining issue.

Reply:
1. Yes, I want to keep these methods isolated from other databases.
2. No table identifiers already preserve the case.
3. Good idea, I'll add comments.

Reply:
Added explanation in docstring.